### PR TITLE
[onboarding] Create proposal for repo deploy if required

### PR DESCRIPTION
**Pull request by sauin**

### CODE REVIEW

**Review by SilkovAlexander — DISMISSED** [review]

File: onboarding-workflow_v5/bin/gosh.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+PROPOSAL_KIND_DEPLOY_REPO=0x000000000000000000000000000000000000000000000000000000000000000c
+
+function get_proposal_details {
+    proposal_addr=$1
+    ensure_provided proposal_addr
+    tonos-cli -j -u network.gosh.sh run $1 getDetails {} \
```
Comment (SilkovAlexander) [inline onboarding-workflow_v5/bin/gosh.sh]: Please move network and abi root to global variables

File: onboarding-workflow_v5/bin/find_proposal.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    -H 'Accept: application/json' \
+    -H 'Connection: keep-alive' \
+    -H 'DNT: 1' \
+    -H 'Origin: https://network.gosh.sh' \
+    --data-binary "{\"query\":\"query { accounts(filter: { code_hash: { eq: \\\"${PROPOSAL_CODE_HASH}\\\" } }) { id } }\"}" \
+    --compressed \
+    | jq -r '.data.accounts[] | .id?'
+)
+
+for account in $accounts; do
```
Comment (SilkovAlexander) [inline onboarding-workflow_v5/bin/find_proposal.sh:28]: This loop checks accounts list once and exits. This logic can be wrong

File: onboarding-workflow_v5/bin/find_proposal.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    if [ -n "$status" ]; then
+        name=$(get_proposal_deploy_repo_name $account)
+        if [ $name == $REPO_NAME ]; then
+            echo "${status}"
+            exit
+        fi
+    else
+        echo ""
+    fi
+done
```
Comment (SilkovAlexander) [inline onboarding-workflow_v5/bin/find_proposal.sh:39]: mb add exit 1 after loop

File: onboarding-workflow_v5/src/eversdk/client.ts [hunk 4]
```diff
@@ -22,7 +22,11 @@ export function getEverClient(): TonClient {
 
         // deno-lint-ignore no-explicit-any
         TonClient.useBinaryLibrary(libWeb as any)
-        everClient = new TonClient()
+        everClient = new TonClient({
+            network: {
+                endpoints: ['network.gosh.sh']
```
Comment (SilkovAlexander) [inline onboarding-workflow_v5/src/eversdk/client.ts]: Endpoint is passed via ENV VARIABLE take it from there

File: onboarding-workflow_v5/src/eversdk/dao.ts [hunk 5]
```diff
@@ -223,3 +230,95 @@ export async function AloneMintDaoReserve(
...
+        if (stdout === "true") {
+            return ProposalStatus.Accepted
+        } else if (stdout === "false") {
+            return ProposalStatus.Rejected
+        } else if (stdout === "null") {
+            return ProposalStatus.InProgress
+        }
+        return ProposalStatus.NonExistent
+    }
+    throw new Error(`Process "${args}" return code ${result.code}\n${stdout}`)
```
Comment (SilkovAlexander) [inline onboarding-workflow_v5/src/eversdk/dao.ts:268]: "returned"

**Review by SilkovAlexander — CHANGES_REQUESTED** [review]

File: onboarding-workflow_v5/bin/find_proposal.sh (hunk 3 above)
Comment (sauin) [reply onboarding-workflow_v5/bin/find_proposal.sh:39]: this return value processed too

**Review by SilkovAlexander — APPROVED** [review]

**Pull request merged**